peerdroid/src/main/java/rdx/works/peerdroid/data/webrtc/WebRtcManager.kt

```diff
@@ -118,46 +118,59 @@ internal class WebRtcManager(applicationContext: Context) {
             this.peerConnection = it
             Timber.d("🔌 created a peer connection: $peerConnection for remote client: $remoteClientId")
         } ?: Timber.e("🔌 failed to create a peer connection")
-        return this.peerConnection
+        return this.peerConnection ?: throw UninitializedPropertyAccessException("property peerConnection has not been initialized")
     }
 
     private fun createRtcDataChannel(remoteClientId: String) {
-        peerConnection.let {
+        peerConnection?.let {
             dataChannel = it.createDataChannel(remoteClientId, dataChannelInit)
             Timber.d("🔌 created a data channel for remote client: $remoteClientId")
         }
     }
 
-    suspend fun createOffer(): Result<SessionDescriptionWrapper.SessionDescriptionValue> =
-        peerConnection.createSuspendingOffer(mediaConstraints = mediaConstraints)
+    suspend fun createOffer(): Result<SessionDescriptionWrapper.SessionDescriptionValue> {
+        return peerConnection?.createSuspendingOffer(
+            mediaConstraints = mediaConstraints
+        ) ?: Result.failure(IllegalStateException("failed to create offer"))
```

Comment:
Is this error logged somewhere?

Reply:
yes this is logged in the `createAndSendAnswerToRemoteClient` of the `PeerdroidLink` 👍🏽